modules/schedule_module/php/appointment.class.inc

```diff
@@ -0,0 +1,261 @@
+     * Handles the duplicateCheck
+     *
+     * @param array $data the request data
+     *
+     * @return void
+     */
+    public function duplicateCheck(array $data)
+    {
+
+        $startAt = $data['AppointmentDate']." ".$data['AppointmentTime'].":00";
+        $factory = \NDB_Factory::singleton();
+        $DB      = $factory->database();
+        $duplicate_check = $DB->pselectRow(
+            "
+           SELECT 
+                * 
+            FROM 
+                appointment 
+            WHERE 
+            SessionID = :sessionId
+            And
+            AppointmentTypeID = :typeId 
+            And
+            StartsAt = :startsAt
+        ",
+            [
+                "sessionId" => $data["Session"],
+                "typeId"    => $data["AppointmentType"],
+                "startsAt"  => $startAt,
+            ]
+        );
+        if (!empty($duplicate_check)) {
+            http_response_code(400);
+            die(
+                json_encode(["error"=>"This appointment already exists."])
+            );
+        }
+    }
+    /**
+     * Handles the appointment adding process
+     *
+     * @param array $data the request data
+     *
+     * @return void
+     */
+    public function addAppointment(array $data): void
+    {
+        $this->duplicateCheck($data);
+        $factory = \NDB_Factory::singleton();
+        $DB      = $factory->database();
+        $startAt = $data['AppointmentDate']." ".$data['AppointmentTime'].":00";
+        // Insert appointment information
+        $DB->insert(
+            "appointment",
+            [
+                "SessionId"         => $data["Session"],
+                "AppointmentTypeId" => $data["AppointmentType"],
+                "StartsAt"          => $startAt,
+            ]
+        );
+
+    }
+    /**
+     * Handles the appointment editing process
+     *
+     * @param array $data the request data
+     *
+     * @return void
+     */
+    public function editAppointment(array $data): void
+    {
+        $this->duplicateCheck($data);
+        $factory = \NDB_Factory::singleton();
+        $DB      = $factory->database();
+        $startAt = $data['AppointmentDate']." ".$data['AppointmentTime'].":00";
+
+        $DB->update(
+            "appointment",
+            [
+                "StartsAt"          => $startAt,
+                "AppointmentTypeID" => $data["AppointmentType"],
+            ],
+            [
+                "AppointmentId" => $data["AppointmentID"],
+            ]
+        );
+
+    }
+    /**
+     * Handles the get all pscid, dccid relative data and session
+     * ex. .../dccid/300001 will return pscid mtl001 and session id
+     * ex. .../pscid/mtl001 will return dccid 300001 and session id
+     *
+     * @param string $type the tpye  (dccid or pscid)
+     * @param string $id   the id   id of dccid or pscid
+     *
+     * @return array all the meta form
+     */
+    public function getMetaForm(string $type,string $id): array
```

Comment:
From what I can tell this method is being used to get a list of visit labels for a single candidate and pass it to the frontend. I'm not sure why the frontend doesn't just use the LORIS API, which includes a list of visits for a candidate